api/python/v2/index.md

```diff
@@ -0,0 +1,173 @@
+
+See below about [API keys](#authentication).
+
+To create a client and connect to a custom Data Commons by a publicly resolvable DNS hostname:
+
+<pre>
+>>> from datacommons_client.client import DataCommonsClient
+>>> client = DataCommonsClient(dc_instance="<var>DNS_HOSTNAME</var>")
+</pre>
+
+For example:
+```python
+client = DataCommonsClient(dc_instance="datacommons.one.org")
+```
+To create a client and connect to a custom Data Commons by a private/non-resolvable address, specify the full API path, including the protocol and API version:
+
+<pre>
+>>> from datacommons_client.client import DataCommonsClient
+>>> client = DataCommonsClient(url="http://<var>YOUR_ADDRESS</var>/core/api/v2/")
+</pre>
+
+For example, to connect to a locally running DataCommons instance:
+
+<pre>
+>>> from datacommons_client.client import DataCommonsClient
+>>> client = DataCommonsClient(url="http://localhost:8080/core/api/v2/")
+</pre>
+
+### Authentication {#authentication}
+
+All access to the base Data Commons (datacommons.org) the V2 APIs must be authenticated and authorized with an API key. The `DataCommonsClient` object manages propagating the API key to all requests, so you don't need to specify it as part of data requests.
+
+We provide a trial API key for general public use. This key will let you try the APIs and make single requests.
+
+<div markdown="span" class="alert alert-secondary" role="alert">
+   <b>Trial key: </b>
+   `AIzaSyCTI4Xz-UW_G2Q2RfknhcfdAnTHq5X5XuI`
+</div>
+
+_The trial key is capped with a limited quota for requests._ If you are planning on using the APIs more rigorously (e.g. for personal or school projects, developing applications, etc.) please request an official key without any quota limits; see [Obtain an API key](/api/index.html#get-key) for information.
+
+For custom DC instances, you do _not_ need to provide any API key.
+
+## Request endpoints and responses
+
+The Python client library sends HTTP POST requests to the Data Commons [REST API endpoints](/api/rest/v2/index.md#service-endpoints) and receives JSON responses. Each endpoint has a corresponding response type. The classes are below:
+
+| API | Endpoint | Description | Response type |
+| --- | --- -----| ----------- |---------------|
+| Observation | [`observation`](observation.md)) | Fetches statistical observations (time series) | `ObservationResponse` |
+| Observations Pandas DataFrame | [`observations_dataframe`](observation.md#) | Same as above, except the functionality is provided by a method of the `DataCommonsClient` class directly, instead of an intermediate endpoint | `pd.DataFrame` |
+| Node | [`node`](node.md) | Fetches information about edges and neighboring nodes | `NodeResponse` |
+| Resolve entities | [`resolve`](resolve.md) | Returns a Data Commons ID ([`DCID`](/glossary.html#dcid)) for entities in the graph | `ResolveResponse` |
+
+To send a request, you inject an endpoint to the client object and call one of the endpoint's methods. For example: 
```

Comment:
The endpoints are all injected on instantiation. I think this instruction would be clearer as:
"To send a request, you use one of the endpoints available as methods of the client object. For example:"

Otherwise, I'd expect for the example to show me how to inject the API client into the endpoint object (something which can be done if we're creating examples of working with the lower-level objects instead of via the DataCommonsClient)

Reply:
Yeah I don't think it's worth adding examples for the individual endpoints. All their functionality would be available from `DataCommonsClient`, so they are more 'internal' than meant to be directly used by users.